src/content/Colors/Colors.tsx

```diff
@@ -1,35 +1,90 @@
 import React from 'react';
 import styled from 'styled-components';
-import { colors, TTextHexColors, IColorSpec } from './../../constants/colors';
+import {
+  colors,
+  IColors,
+  IBrandColors,
+  IActionColors,
+  INeutralColors,
+  INotificationColors,
+} from './../../constants/colors';
 
-export type TColorNames = keyof typeof colors.base | keyof typeof colors.neutral | keyof typeof colors.semantic;
+type TColorVariants = keyof IColors;
 
 interface ISColorProps {
-  color: TTextHexColors;
-  colorName: TColorNames;
+  color: string;
+  colorName: TColorVariants;
+}
+
+interface IColorsProps {
+  /**
+   * section of the default colors
+   * @ignore
+   */
+  variant: 'brand' | 'actions' | 'neutral' | 'notifications';
+}
+
+interface IColorGroups {
+  brand: Array<keyof IBrandColors>;
+  actions: Array<keyof IActionColors>;
+  neutral: Array<keyof INeutralColors>;
+  notifications: Array<keyof INotificationColors>;
 }
 
 const SColors = styled.div`
   display: flex;
   flex-flow: row wrap;
 `;
 
-const mapColorToFontColor: { [K in TColorNames]: 'white' | 'black' } = {
-  primary: 'white',
-  secondary: 'white',
+const mapColorToFontColor: { [C in TColorVariants]: 'white' | 'black' } = {
```

Comment:
`const mapColorToFontColor: Record<TColorVariants, 'white' | 'black'> = {`

Reply:
Is there a way to detect if colour is closer to white or black? Then make this automagically? Looks like [there are some ways](https://stackoverflow.com/a/58427960/1762002). It's more a nice to have, I mean I understand this is not a priority.

Reply:
I'll give a try 🙃

Reply:
@poteirard these implementations feel a bit magical ( _unless you know the inner workings of `hex`and `rgb`_ ) and if there's an issue with them debugging won't be fun 😓 

The current verbose solution in `mapColorToFontColor` is ugly but quite **easy to debug and control** ( _as well as for handling special cases like_ `colors.action` ).

So thinking to keep it instead, what's your feeling?